src/types/qwdrmleasev1manager.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2023 rewine <luhongxu@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#include "qwdrmleasev1.h"
+#include "qwdisplay.h"
+#include "qwbackend.h"
+#include "qwoutput.h"
+#include "util/qwsignalconnector.h"
+
+#include <QHash>
+
+extern "C" {
+#include <wlr/types/wlr_drm_lease_v1.h>
+}
+
+QW_BEGIN_NAMESPACE
+
+class QWDrmLeaseV1ManagerPrivate : public QWObjectPrivate
+{
+public:
+    QWDrmLeaseV1ManagerPrivate(wlr_drm_lease_v1_manager *handle, bool isOwner, QWDrmLeaseV1Manager *qq)
+        : QWObjectPrivate(handle, isOwner, qq)
+    {
+        Q_ASSERT(!map.contains(handle));
+        map.insert(handle, qq);
+        sc.connect(&handle->events.request, this, &QWDrmLeaseV1ManagerPrivate::on_request);
+    }
+    ~QWDrmLeaseV1ManagerPrivate() {
+        if (!m_handle)
+            return;
+        destroy();
+    }
+
+    inline void destroy() {
+        Q_ASSERT(m_handle);
+        Q_ASSERT(map.contains(m_handle));
+        Q_EMIT q_func()->beforeDestroy(q_func());
+        map.remove(m_handle);
+        sc.invalidate();
+    }
+
+    void on_request(void *);
+
+    static QHash<void*, QWDrmLeaseV1Manager*> map;
+    QW_DECLARE_PUBLIC(QWDrmLeaseV1Manager)
+    QWSignalConnector sc;
+};
+QHash<void*, QWDrmLeaseV1Manager*> QWDrmLeaseV1ManagerPrivate::map;
+
+void QWDrmLeaseV1ManagerPrivate::on_request(void *data)
+{
+    auto *drmLeaseRequest = QWDrmLeaseRequestV1::from(reinterpret_cast<wlr_drm_lease_request_v1*>(data));
+    Q_EMIT q_func()->request(drmLeaseRequest);
+}
+
+QWDrmLeaseV1Manager::QWDrmLeaseV1Manager(wlr_drm_lease_v1_manager *handle, bool isOwner)
+    : QObject(nullptr)
```

Comment:
The kind of wlr protocol object is not offer `destroy` signal, it's following `wl_display` to destroy, so we can make the `QW*` object to following `QWDisplay` to destroy, set its parent to QWDisplay at here.

Reply:
And modify the exists same kind class in a new commit.